git-hub

```diff
@@ -1471,6 +1474,143 @@ class PullCmd (IssueCmd):
+                    "to configure a diff tool\n")
+                die("No diff tool has been configured. Aborting.")
+
+            # BUG: using vimdiff as the diff tool hangs the terminal :(
+            # The reason is most likely that vimdiff doesn't open in a new
+            # window altogether. For now, we just reject vimdiff.
+            if git_config('diff.tool', prefix='') == 'vimdiff':
+                die("vimdiff cannot be used as the diff tool because of a "
+                    "known issue (please consider using gvimdiff instead).")
+
+            # First confirm that there are no modified files
+            modified_files = git('diff', '--name-only')
+            if modified_files:
+                die("Modified files detected. Aborting.")
+
+            # Next confirm that the base is valid
+            try:
+                git_quiet(1, 'rev-parse', '--short', '--verify', args.base)
+            except GitError:
+                die("Base '{}' doesn't exist. Aborting.", args.base)
+
+            # Warn if the PR is already closed
+            pull = req.get(cls.url(args.pull))
+            if pull['state'] == 'closed':
+                warnf('Reviewing a closed pull request '
+                    '(closed at {closed_at})!', **pull)
+
+            remote_url = pull['head']['repo'][config.urltype]
+            remote_branch = pull['head']['ref']
+
+            infof("Fetching '{}' from {}", remote_branch, remote_url)
+            git_quiet(1, 'fetch', remote_url, remote_branch)
+
+            local_branch = 'pull_' + args.pull
+
+            # Warn if the pull branch already exists locally
+            try:
+                pull_branch_hash = git_quiet(1, 'rev-parse', '--short',
+                    '--verify', local_branch)
+                warnf("Pull branch '{}' already exists at commit '{}'. "
+                    "It _may_ get moved.", local_branch, pull_branch_hash)
+                git_quiet(1, 'checkout', '-B', local_branch, 'FETCH_HEAD')
+            except GitError:
+                infof("Checking out locally as branch '{}'", local_branch)
+                git_quiet(1, 'checkout', '-b', local_branch, 'FETCH_HEAD')
+
+            commits_range = args.base + '..' + local_branch
+            commits_list = git('log', '--oneline', commits_range).split('\n')
+            infof("Found {} commits in range '{}'",
+                len(commits_list), commits_range)
+
+            if args.auto_skip:
+                if args.auto_skip >= len(commits_list):
+                    infof('All commits automatically skipped')
+                    return
+                else:
+                    if args.auto_skip == 1:
+                        infof('Skipping commit 1')
+                    elif args.auto_skip == 2:
+                        infof('Skipping commits 1 & 2')
+                    else:
+                        infof('Skipping commits 1 to {}', args.auto_skip)
```

Comment:
Why would you tell the user he used an option he just wrote himself in the command-line? :)

At best this should be `debugf()`.